contracts/token/ERC20/extensions/draft-ERC20TemporaryApproval.sol

```diff
@@ -0,0 +1,106 @@
+     */
+    function _temporaryAllowance(address owner, address spender) internal view virtual returns (uint256) {
+        return ERC20TemporaryApprovalStorageLocation.deriveMapping(owner).deriveMapping(spender).asUint256().tload();
+    }
+
+    /**
+     * @dev Alternative to {approve} that sets a `value` amount of tokens as the temporary allowance of `spender` over
+     * the caller's tokens.
+     *
+     * Returns a boolean value indicating whether the operation succeeded.
+     *
+     * Requirements:
+     * - `spender` cannot be the zero address.
+     *
+     * Does NOT emit an {Approval} event.
+     */
+    function temporaryApprove(address spender, uint256 value) public virtual returns (bool) {
+        _temporaryApprove(_msgSender(), spender, value);
+        return true;
+    }
+
+    /**
+     * @dev Sets `value` as the temporary allowance of `spender` over the `owner` s tokens.
+     *
+     * This internal function is equivalent to `temporaryApprove`, and can be used to e.g. set automatic allowances
+     * for certain subsystems, etc.
+     *
+     * Requirements:
+     * - `owner` cannot be the zero address.
+     * - `spender` cannot be the zero address.
+     *
+     * Does NOT emit an {Approval} event.
+     */
+    function _temporaryApprove(address owner, address spender, uint256 value) internal virtual {
+        if (owner == address(0)) {
+            revert ERC20InvalidApprover(address(0));
+        }
+        if (spender == address(0)) {
+            revert ERC20InvalidSpender(address(0));
+        }
+        ERC20TemporaryApprovalStorageLocation.deriveMapping(owner).deriveMapping(spender).asUint256().tstore(value);
+    }
+
+    /**
+     * @dev {_spendAllowance} override that consumes the temporary allowance (if any) before eventually falling back
+     * to consumming the persistent allowance.
+     */
+    function _spendAllowance(address owner, address spender, uint256 value) internal virtual override {
+        unchecked {
+            // load transient allowance
+            uint256 currentTemporaryAllowance = _temporaryAllowance(owner, spender);
+            // if there is temporary allowance
+            if (currentTemporaryAllowance > 0) {
+                // if infinite, do nothing
+                if (currentTemporaryAllowance == type(uint256).max) return;
+                // check how much of the value is covered by the transient allowance
+                uint256 spendTemporaryAllowance = Math.min(currentTemporaryAllowance, value);
+                // decrease transient allowance accordingly
+                _temporaryApprove(owner, spender, currentTemporaryAllowance - spendTemporaryAllowance);
+                // update value necessary
+                value -= spendTemporaryAllowance;
+            }
+            // reduce any remaining value from the persistent allowance
+            super._spendAllowance(owner, spender, value);
+        }
```

Comment:
This is safe, but it's wrapping a call to `allowance` in ERC20.sol
I'm worried that overriding `allowance` is not a crazy use case, and this extension would inadvertently uncheck its operations.

```suggestion
        }
        super._spendAllowance(owner, spender, value);
```

Reply:
> I'm worried that overriding allowance is not a crazy use case, and this extension would inadvertently uncheck its operations.

Not sure I understand what you mean. You know that doing 
```
unchecked { fn1(a, b); }
``` 
does not propagate unchecked to chat happens inside `fn1` ?

Reply:
I didn't know! I was always manually make sure propagated `unchecked` blocks were not an issue.
Solving